libmscore/textedit.cpp

```diff
@@ -260,8 +260,19 @@ bool TextBase::edit(EditData& ed)
                         return true;
 
                   case Qt::Key_Delete:
-                        if (!deleteSelectedText(ed))
-                              score()->undo(new RemoveText(_cursor, QString(_cursor->currentCharacter())), &ed);
+                        if (!deleteSelectedText(ed)) {
+                              // check for move down
+                              if (_cursor->column() == _cursor->columns() && _cursor->column() != 0) { // if you are on the end of the line, delete the newline char
+                                    int cursorRow = _cursor->row();
+                                    _cursor->movePosition(QTextCursor::Down);
+                                    if (_cursor->row() != cursorRow) {
+                                          _cursor->movePosition(QTextCursor::StartOfLine);
+                                          score()->undo(new JoinText(_cursor), &ed);
+                                          }
+                                    }
+                              else
+                                    score()->undo(new RemoveText(_cursor, QString(_cursor->currentCharacter())), &ed);
```

Comment:
Even though this is just a single statement, better add braces, because the if branch has braces. As per https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules#Braces

Reply:
Done.